### PR TITLE
MAYA-114593 preserve load state of prims

### PR DESCRIPTION
**Pull request by pierrebai-adsk**
Added code to preserve the loaded or unloaded state of a prim when grouped, reparented or renamed. Note that we don't remove the old rules when duplicating since the source prim still exists.

- Refactor the existing load-state preservation code in reusable functions.
- Use them in the duplicate, rename, and reparenting commands.
- Add unit tests for grouping and renaming, which also covers parenting since grouping reparents.

### CODE REVIEW

File: lib/mayaUsd/utils/loadRules.cpp [hunk 1]
```diff
@@ -0,0 +1,106 @@
...
+    PXR_NS::UsdStage&      stage,
+    const PXR_NS::SdfPath& fromPath,
+    const PXR_NS::SdfPath& destPath)
+{
+    // Note: get a *copy* of the rules since we are going to insert new rules as we iterate.
+    auto loadRules = stage.GetLoadRules();
+
+    // Retrieve the effective rule for the source path.
+    //
+    // The reason we retrieve the effetive rule is that even
```
Comment (ppt-adsk) [inline lib/mayaUsd/utils/loadRules.cpp:31]: "effetive" --> "effective".

File: lib/mayaUsd/utils/loadRules.cpp [hunk 2]
```diff
@@ -0,0 +1,106 @@
...
+    // The reason we don't simply add this specific rule for all cases and
+    // avoid the above work is that sub-paths might have rules and we need
+    // to preserve those rules. So we would need to do the above work anyway.
+    //
+    // Given that, a common case is that we don't need to add an additional
+    // rule. Always adding it would add unnecessary rules.
+    //
+    // Note: the UsdStageLoadRules has a Minimize function that simplifies
+    // rules, but we don't want to change rules the user might have set.
+    // The user may expect those rules to exists for some future purpose
```
Comment (ppt-adsk) [inline lib/mayaUsd/utils/loadRules.cpp:75]: "to exists" --> "to exist".

**Review by ppt-adsk — APPROVED** [review]
Great job, just minor typos and grammar.

**Review by ppt-adsk — APPROVED** [review]

**Comment by pierrebai-adsk** [comment]
Only the known flaky test TestALUSDMayaPython_LayerManager failed in the interactive linux instance.

**Pull request merged**